web/src/main/java/se/inera/intyg/webcert/web/service/message/MessageImportServiceImpl.java

```diff
@@ -0,0 +1,366 @@
+        if (isSendMessageToCare(messageFromIT.getMessageContent())) {
+            processIncomingMessage(messageFromIT, certificate);
+        } else if (isSendMessageToRecipient(messageFromIT.getMessageContent())) {
+            processOutgoingMessage(messageFromIT, certificate);
+        } else {
+            LOG.error("Couldn't process message. Unknown message format for messageId: " + messageFromIT.getMessageId());
+        }
+    }
+
+    private void processOutgoingMessage(MessageFromIT messageFromIT, IntygContentHolder certificate) {
+        final var messageToRecipient = convertXmlToSendMessageToRecipient(messageFromIT.getMessageContent(), messageFromIT.getMessageId());
+        if (messageToRecipient == null) {
+            return;
+        }
+
+        final var convertedMessage = convertMessageToRecipient(messageToRecipient, certificate);
+        final var savedMessage = messageRepository.save(convertedMessage);
+
+        if (isMessageAnAnswer(savedMessage)) {
+            updateAnsweredMessage(savedMessage, Status.CLOSED);
+        }
+
+        monitoringLogService.logMessageImported(savedMessage.getIntygsId(), savedMessage.getMeddelandeId(),
+            certificate.getUtlatande().getGrundData().getSkapadAv().getVardenhet().getVardgivare().getVardgivarid(),
+            savedMessage.getEnhetId(), isMessageAnAnswer(savedMessage) ? "SendMessageToRecipient - Answer" : "SendMessageToRecipient");
+    }
+
+    private void processIncomingMessage(MessageFromIT messageFromIT, IntygContentHolder certificate) {
+        final var messageToCare = convertXmlToSendMessageToCare(messageFromIT.getMessageContent(), messageFromIT.getMessageId());
+        if (messageToCare == null) {
+            return;
+        }
+
+        final var convertedMessage = convertMessageToCare(messageToCare, certificate);
+        final var savedMessage = messageRepository.save(convertedMessage);
+
+        if (isMessageAnAnswer(savedMessage)) {
+            updateAnsweredMessage(savedMessage, Status.CLOSED);
+        }
+
+        if (isMessageReminder(savedMessage)) {
+            updatedRemindedMessage(savedMessage);
+        }
+
+        if (savedMessage.getStatus() == Status.PENDING_INTERNAL_ACTION && savedMessage.getAmne() == ArendeAmne.KOMPLT) {
+            if (isComplemented(certificate)) {
+                savedMessage.setSenasteHandelse(certificate.getRelations().getLatestChildRelations().getComplementedByIntyg().getSkapad());
+                savedMessage.setStatus(Status.CLOSED);
+            }
+        }
+
+        monitoringLogService.logMessageImported(savedMessage.getIntygsId(), savedMessage.getMeddelandeId(),
+            certificate.getUtlatande().getGrundData().getSkapadAv().getVardenhet().getVardgivare().getVardgivarid(),
+            savedMessage.getEnhetId(), isMessageAnAnswer(savedMessage) ? "SendMessageToCare - Answer" : "SendMessageToCare");
+    }
+
+    private void updateAnsweredMessage(Arende message, Status answeredMessageStatus) {
```

Comment:
Ska inte metoden spara till databasen?`Eller behövs inte det pga @Transactional(propagation = Propagation.REQUIRES_NEW)?

Reply:
Uppdateringar på entiteten sparas automatiskt och comittas när transaktionen är klar.

https://stackoverflow.com/questions/13588565/spring-data-jpa-why-are-changes-to-a-returned-entity-automatically-persisted

Reply:
Jaha. Se där!